docs/data/data-grid/layout/FlexLayoutGrid.tsx

```diff
@@ -11,7 +11,7 @@ export default function FlexLayoutGrid() {
 
   return (
     <div style={{ height: 400, width: '100%' }}>
-      <div style={{ display: 'flex', height: '100%' }}>
+      <div style={{ display: 'flex', flexDirection: 'column', height: '100%' }}>
         <div style={{ flexGrow: 1 }}>
           <DataGrid {...data} />
```

Comment:
I wonder what is the point of this demo.
There are 3 wrapper divs, 2 of which seem to be redundant since it's equivalent of
```tsx
<div style={{ height: 400, width: '100%' }}>
  <DataGrid {...data} />
</div>
```

What are we trying to cover in this section? https://mui.com/x/react-data-grid/layout/#flex-layout

Reply:
@m4theushw could you take a look at this one? It's off-topic, but it's not clear for me what's the intention of that demo

Reply:
If I recall correctly, I used https://ag-grid.com/javascript-data-grid/grid-size/#inside-flexbox-container as inspiration for this demo. But in hindsight, it feels that developers learn nothing from https://mui.com/x/react-data-grid/layout/#flex-layout. It could be good to delete it. This would make more sense

<img width="781" alt="Screenshot 2023-01-08 at 22 25 57" src="https://user-images.githubusercontent.com/3165635/211219902-7aa82972-61a2-45c3-87a2-cca360ed301f.png">

but really? Do we need to teach developer CSS Grid/Flexbox?

Reply:
Thanks, @oliviertassinari 
Few minor concerns I also had about the flex demo:
1. Developers might think that a flex container is required for the grid to work (especially on a quick look at the docs, since it's the first demo on the page).
2. Developers might copy the whole snippet with 2 nested divs, which is ... unnecessarily complicated 🤷🏻‍♂️

> but really? Do we need to teach developer CSS Grid/Flexbox?

I think we can remove the whole section. The main requirement for the parent container is covered in the first few sentences anyway

Reply:
It sounds like this is fixed now 👍 

<img width="561" alt="Screenshot 2023-01-29 at 16 30 37" src="https://user-images.githubusercontent.com/3165635/215336736-a6a79acd-6ed0-40ff-a6f9-8939d896a56d.png">

https://mui-org.slack.com/archives/C041SDSF32L/p1674915288809749